### DISCUSSION TITLE
randomize_question_order=true in survey-likert trial

### DISCUSSION THREAD

**Discussion by sonhkim** [jspsych/jsPsych#1079] [Q&A]
Hello, I'm building a jsPsych experiment where a participant answers ~100 Likert-scale questions. I want to randomize the order of those 100 questions for each participant. I have the following code, but it doesn't do the randomization that it is supposed to do: 

`var question_trial = {
      type: 'survey-likert',
      questions: function() {
      	var prompts = [
      	{
      		prompt: jsPsych.timelineVariable("question", true),
      		name: jsPsych.timelineVariable("verb", true),
      		labels: scale, 
      		required: true
      	},
      	];
      	return prompts;
      },
      randomize_question_order: true,
      };
 
var survey_procedure = {
			timeline: [question_trial],
			timeline_variables: stims
		}
  timeline.push(survey_procedure);
`
(I have the var "stims" written in a separate js file, which is just a list of dictionaries)

If I add the following line in the code, then it does randomize the order of questions: 
stims = jsPsych.randomization.shuffle(stims)  

So, I got to make this work, apparently, but I want to understand why the 'randomize_question_order' parameter didn't do a thing.

**Comment by becky-gilbert** [comment]
It looks like you're showing a single question on each page, is that right? If so, the reason that this doesn't work the way you expect is because the `randomize_question_order: true` option in the survey plugins controls the question order when there are multiple questions on the _same page_. Whereas what I think you're doing is using timeline variables to repeat your `question_trial` lots of times, with one question per page. In this case, the question order is determined by the way that you set up the timeline variables (i.e. your `survey_procedure` object). 

If you want to check whether this is the case, you could remove the line where you shuffle the stims array and add the [timeline variables randomize order option](https://www.jspsych.org/overview/timeline/#random-orders-of-trials) to your `survey_procedure`:
```javascript
var survey_procedure = {
  timeline: [question_trial],
  timeline_variables: stims,
  randomize_order: true
}
```
